Core/contentblocker.js

```diff
@@ -553,29 +553,39 @@ _utf8_encode : function (string) {
     // Init
     (function() {
         
-        duckduckgoDebugMessaging.log("installing beforeload detection")
-        document.addEventListener("beforeload", function(event) {
-
-            if (event.target.nodeName == "LINK") {
-                type = event.target.rel
-            } else if (event.target.nodeName == "IMG") {
-                type = "image"
-            } else if (event.target.nodeName == "IFRAME") {
-                type = "subdocument"
-            } else {
-                type = event.target.nodeName
-            }
-
-            duckduckgoDebugMessaging.log("checking " + event.url + " (" + type + ")");
-            if (shouldBlock(event.url, type)) {
-                duckduckgoDebugMessaging.log("blocking beforeload")
-                event.preventDefault()
-                event.stopPropagation()
-            } else {
-                duckduckgoDebugMessaging.log("don't block " + event.url);
-                return
-            }
-        }, true)
+        duckduckgoDebugMessaging.log("installing load detection")
+        window.addEventListener("load", function(event) {
+            [].slice.apply(document.scripts).forEach(function(el) {
```

Comment:
Does Safari support `[...document.scripts]` here?

Reply:
Not sure. This is in line with other parts of this codebase though. Bringing the style of the JS up to standard is out of scope here though and has been planned as a future project.